File: app/src/main/java/com/lumiyaviewer/lumiya/res/text/DrawableTextCache.java
package com.lumiyaviewer.lumiya.res.text;

import com.lumiyaviewer.lumiya.res.ResourceManager;
import com.lumiyaviewer.lumiya.res.ResourceMemoryCache;
import com.lumiyaviewer.lumiya.res.ResourceRequest;
import com.lumiyaviewer.lumiya.res.executors.PrimComputeExecutor;

public class DrawableTextCache extends ResourceMemoryCache<DrawableTextParams, DrawableTextBitmap> {
    private final int fontSize;

    private static class TextGenRequest extends ResourceRequest<DrawableTextParams, DrawableTextBitmap> implements Runnable {
        private final int fontSize;

        TextGenRequest(DrawableTextParams drawableTextParams, int i, ResourceManager<DrawableTextParams, DrawableTextBitmap> resourceManager) {
            super(drawableTextParams, resourceManager);
            this.fontSize = i;
        }

        public void execute() {
            PrimComputeExecutor.getInstance().execute(this);
        }

        public void run() {
            completeRequest(new DrawableTextBitmap((DrawableTextParams) getParams(), this.fontSize));
        }
    }

    public DrawableTextCache(int i) {
        this.fontSize = i;
    }

    protected ResourceRequest<DrawableTextParams, DrawableTextBitmap> CreateNewRequest(DrawableTextParams drawableTextParams, ResourceManager<DrawableTextParams, DrawableTextBitmap> resourceManager) {
        return new TextGenRequest(drawableTextParams, this.fontSize, resourceManager);
    }
}

File: lumiya/src/main/java/com/lumiyaviewer/lumiya/res/text/DrawableTextCache.kt
package com.lumiyaviewer.lumiya.res.text

import com.lumiyaviewer.lumiya.res.ResourceManager
import com.lumiyaviewer.lumiya.res.ResourceMemoryCache
import com.lumiyaviewer.lumiya.res.ResourceRequest
import com.lumiyaviewer.lumiya.res.executors.PrimComputeExecutor
import com.lumiyaviewer.lumiya.memory.MemoryManager

class DrawableTextCache(private val fontSize: Int) : ResourceMemoryCache<DrawableTextParams, DrawableTextBitmap>(MemoryManager()) {

    private inner class TextGenRequest(
        params: DrawableTextParams,
        private val fontSize: Int,
        manager: ResourceManager<DrawableTextParams, DrawableTextBitmap>
    ) : ResourceRequest<DrawableTextParams, DrawableTextBitmap>(params, manager), Runnable {

        override fun execute() {
            PrimComputeExecutor.getInstance().execute(this)
        }

        override fun run() {
            completeRequest(DrawableTextBitmap(getParams(), this.fontSize))
        }
    }

    override fun CreateNewRequest(
        params: DrawableTextParams,
        manager: ResourceManager<DrawableTextParams, DrawableTextBitmap>
    ): ResourceRequest<DrawableTextParams, DrawableTextBitmap> {
        return TextGenRequest(params, this.fontSize, manager)
    }
}
